src/PackageUtilities.js

```diff
@@ -27,25 +28,25 @@ export default class PackageUtilities {
     return require(PackageUtilities.getPackageConfigPath(packagesPath, name));
   }
 
-  static getPackages(packagesPath) {
+  static getPackages(repository) {
     const packages = [];
 
-    FileSystemUtilities.readdirSync(packagesPath).forEach((packageDirectory) => {
-      if (packageDirectory[0] === ".") {
-        return;
-      }
+    repository.packageConfigs.forEach(({glob}) => {
 
-      const packagePath = PackageUtilities.getPackagePath(packagesPath, packageDirectory);
-      const packageConfigPath = PackageUtilities.getPackageConfigPath(packagesPath, packageDirectory);
+      globSync(glob)
+        .map((fn) => path.resolve(fn))
+        .forEach((packageConfigPath) => {
+          const packagePath = path.dirname(packageConfigPath);
 
```

Comment:
The section below just got indented a bit.  Easier to view with a [no whitespace diff](https://github.com/lerna/lerna/pull/332/files?w=1).